apollo-spaceport/proto/agents.proto

```diff
@@ -0,0 +1,31 @@
+syntax = "proto3";
+
+import "proto/reports.proto";
+
+package Agent;
+
+service Reporter {
```

Comment:
This overall structure doesn't leave an obvious place to add `referenced_fields_by_type` later. Perhaps just having one endpoint that takes a TracesAndStats (which contains Report.ContextualizedStats and Report.Trace too) would be better? That said, if this protocol is considered to be internal to router and is open to change at any time, then perhaps don't worry about this until implementing `referenced_fields_by_type`.

... but I think you should at least have a comment somewhere in the codebase tracking the fact that referenced_fields_by_type is not yet implemented. 

Note that this feature is something else that you might want to start off by loading over the query planner bridge interface.  The questions "does it validate", "what fields does it return to", and "what is the query plan" are all basic facts about the request that can all be cached with the exact same lifetime and they overlap in what they need to do to calculate it.

Reply:
Seems like the router_bridge is going to solve many problems for this! :) It is a router internal protocol (at least for now), so I'm not too concerned about evolving it. I'll open an issue for adding "referenced..."